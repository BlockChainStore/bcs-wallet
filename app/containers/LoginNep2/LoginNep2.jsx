// @flow
import React, { Component } from 'react'

import Button from '../../components/Button'
import HomeButtonLink from '../../components/HomeButtonLink'
import PasswordField from '../../components/PasswordField'

import loginStyles from '../../styles/login.scss'

type Props = {
  loginNep2: Function,
  history: Object
}

type State = {
  encryptedWIF: string,
  passphrase: string
}

export default class LoginNep2 extends Component<Props, State> {
  state = {
    encryptedWIF: '',
    passphrase: ''
  }

  render () {
    const { loginNep2, history } = this.props
    const { encryptedWIF, passphrase } = this.state
    const loginButtonDisabled = encryptedWIF === '' || passphrase === ''

    return (
      <div id='loginPage' className={loginStyles.loginPage}>
        <div className={loginStyles.title}>Login using an encrypted key:</div>
        <form onSubmit={(e) => { e.preventDefault(); loginNep2(passphrase, encryptedWIF, history) }}>
          <div className={loginStyles.loginForm}>
            <PasswordField
              placeholder='Enter your passphrase here'
              onChange={(e) => this.setState({ passphrase: e.target.value })}
              value={passphrase}
              autoFocus
            />
            <PasswordField
              placeholder='Enter your encrypted key here'
              onChange={(e) => this.setState({ encryptedWIF: e.target.value })}
              value={encryptedWIF}
            />
          </div>
          <div>
            <Button
              id='loginButton'
              type='submit'
              disabled={loginButtonDisabled}>Login</Button>
            <HomeButtonLink />
          </div>
        </form>
      </div>
    )
  }
}
