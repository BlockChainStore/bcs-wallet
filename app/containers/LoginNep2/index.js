// @flow
import { compose } from 'recompose'

import LoginNep2 from './LoginNep2'
import withActions from '../../hocs/api/withActions'
import withFailureNotification from '../../hocs/withFailureNotification'
import { nep2LoginActions } from '../../actions/authActions'

const mapActionsToProps = (actions) => ({
  loginNep2: (passphrase, encryptedWIF) => actions.request({ passphrase, encryptedWIF })
})

export default compose(
  withActions(nep2LoginActions, mapActionsToProps),
  withFailureNotification(nep2LoginActions)
)(LoginNep2)
