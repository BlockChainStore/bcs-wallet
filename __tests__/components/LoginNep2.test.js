import React from 'react'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import thunk from 'redux-thunk'
import { shallow, mount } from 'enzyme'
import { createMemoryHistory } from 'history'

import LoginNep2 from '../../app/containers/LoginNep2'

jest.useFakeTimers()
jest.mock('neon-js')

const setup = (shallowRender = true) => {
  const store = configureStore([thunk])({})

  let wrapper
  if (shallowRender) {
    wrapper = shallow(<LoginNep2 store={store} />)
  } else {
    const history = createMemoryHistory('/dashboard')
    wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <LoginNep2 history={history} />
        </BrowserRouter>
      </Provider>
    )
  }

  return {
    store,
    wrapper
  }
}

describe('LoginNep2', () => {
  test('renders without crashing', (done) => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
    done()
  })
  test('renders correctly with initial state', (done) => {
    const { wrapper } = setup(false)

    const fields = wrapper.find('input[type="password"]')
    expect(fields.length).toEqual(2)

    const passwordField = fields.get(0)
    const keyField = fields.get(1)

    expect(passwordField.props.value).toEqual('')
    expect(passwordField.props.placeholder).toEqual('Enter your passphrase here')
    expect(passwordField.props.type).toEqual('password')
    expect(keyField.props.value).toEqual('')
    expect(keyField.props.placeholder).toEqual(('Enter your encrypted key here'))
    expect(keyField.props.type).toEqual('password')
    done()
  })

  test('the login button is working correctly with no passphrase or wif', (done) => {
    const { wrapper, store } = setup(false)

    wrapper.find('#loginButton').first().simulate('click')
    Promise.resolve('pause').then(() => {
      const actions = store.getActions()
      expect(actions.length).toEqual(0)
      done()
    })
  })

  test('the login button is working correctly with key and passphrase', () => {
    const { wrapper, store } = setup(false)

    const passwordField = wrapper.find('input[placeholder="Enter your passphrase here"]')
    passwordField.instance().value = 'Th!s1$@FakePassphrase'
    passwordField.simulate('change')

    const keyField = wrapper.find('input[placeholder="Enter your encrypted key here"]')
    keyField.instance().value = '6PYUGtvXiT5TBetgWf77QyAFidQj61V8FJeFBFtYttmsSxcbmP4vCFRCWu'
    keyField.simulate('change')

    wrapper.find('#loginButton').first().simulate('submit')

    const actions = store.getActions()
    expect(actions.length).toEqual(1)
    expect(actions[0].type).toEqual('AUTH/REQ/REQUEST')
  })
})
