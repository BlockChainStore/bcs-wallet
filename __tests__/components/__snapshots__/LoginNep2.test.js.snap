// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginNep2 renders without crashing 1`] = `
<withActions(Connect(Connect(withError(Connect(withProgressProp(ErrorNotifier))))))
  loginNep2={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
